react-components/src/components/RevealToolbar/LayersContainer/CadModelLayersContainer.tsx

```diff
@@ -2,38 +2,29 @@
  * Copyright 2023 Cognite AS
  */
 
-import React, { type ReactElement, useState } from 'react';
+import React, { type ReactElement } from 'react';
 import { useReveal } from '../../RevealContainer/RevealContext';
 import { type CogniteCadModel } from '@cognite/reveal';
 import { Checkbox, Flex, Menu } from '@cognite/cogs.js';
 import { StyledChipCount, StyledLabel, StyledSubMenu } from './elements';
-import { use3DModelName } from '../../../hooks/use3DModelName';
 import uniqueId from 'lodash/uniqueId';
+import { type Reveal3DResourcesLayersProps } from './types';
 
-export const CadModelLayersContainer = (): ReactElement => {
+export const CadModelLayersContainer = ({
+  layerProps
+}: {
+  layerProps: Reveal3DResourcesLayersProps;
+}): ReactElement => {
   const viewer = useReveal();
-  const cadModels = viewer.models.filter((model) => model.type === 'cad');
-  const cadModelIds = cadModels.map((model) => model.modelId);
 
-  const modelName = use3DModelName(cadModelIds);
+  const { cadModels } = layerProps.reveal3DResourcesStates;
 
-  const [selectedCadModels, setSelectedCadModels] = useState<
-    Array<{ model: CogniteCadModel; isToggled: boolean; name: string }>
-  >(
-    cadModels.map((model, index) => ({
-      model: model as CogniteCadModel,
-      isToggled: (model as CogniteCadModel).visible,
-      name: modelName?.data?.[index] ?? 'No model name'
-    }))
-  );
-
-  const [allCadModelVisible, setAllCadModelVisible] = useState(true);
-  const [indeterminate, setIndeterminate] = useState<boolean>(false);
-
-  const count = selectedCadModels.length.toString();
+  const count = cadModels.length.toString();
+  const allModelVisible = !cadModels.every((data) => !data.isToggled);
```

Comment:
Not sure I follow the logic of this - `allModelVisible` is the same as `not every cadmodel is not toggled`? Same with the other model types too